include/xtensor-io/xblosc.hpp

```diff
@@ -0,0 +1,193 @@
+            char* compressed_buffer = new char[max_compressed_size];
+            if (!blosc_initialized)
+            {
+                blosc_init();
+                blosc_initialized = true;
+            }
+            int true_compressed_size = blosc_compress(clevel, doshuffle, sizeof(value_type), uncompressed_size, uncompressed_buffer, compressed_buffer, max_compressed_size);
+            if (true_compressed_size == 0)
+            {
+                throw std::runtime_error("buffer is uncompressible");
+            }
+            else if (true_compressed_size < 0)
+            {
+                throw std::runtime_error("compression error");
+            }
+            stream.write(compressed_buffer,
+                         std::streamsize(true_compressed_size));
+            delete[] compressed_buffer;
+        }
+    }  // namespace detail
+
+    /**
+     * Save xexpression to blosc format
+     *
+     * @param stream An output stream to which to dump the data
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline void dump_blosc(std::ofstream& stream, const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+    }
+
+    /**
+     * Save xexpression to blosc format
+     *
+     * @param filename The filename or path to dump the data
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline void dump_blosc(const std::string& filename, const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        std::ofstream stream(filename, std::ofstream::binary);
+        if (!stream.is_open())
+        {
+            std::runtime_error("IO Error: failed to open file");
+        }
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+    }
+
+    /**
+     * Save xexpression to blosc format in a string
+     *
+     * @param e the xexpression
+     */
+    template <typename E>
+    inline std::string dump_blosc(const xexpression<E>& e, int clevel=5, int doshuffle=1)
+    {
+        std::stringstream stream;
+        detail::dump_blosc_stream(stream, e, clevel, doshuffle);
+        return stream.str();
+    }
+
+    /**
+     * Loads a blosc file
+     *
+     * @param stream An input stream from which to load the file
+     * @tparam T select the type of the blosc file
+     * @tparam L select layout_type::column_major if you stored data in
+     *           Fortran format
+     * @return xarray with contents from blosc file
+     */
+    template <typename T, layout_type L = layout_type::dynamic>
+    inline auto load_blosc(std::istream& stream)
+    {
+        std::size_t uncompressed_size;
+        T* uncompressed_buffer = reinterpret_cast<T*>(detail::load_blosc_file(stream, uncompressed_size));
+        std::size_t size = uncompressed_size / sizeof(T);
+        std::vector<std::size_t> shape = {size};
+        auto array = adapt(uncompressed_buffer, size, no_ownership(), shape);
```

Comment:
The adaptor should acquire the ownerhsip of uncompressed_buffer, otherwise it will leak.

Reply:
But then it will be freed twice? If I change it to:
```cpp
auto array = adapt(uncompressed_buffer, size, acquire_ownership(), shape);
```
I get the following error:
```
double free or corruption (out)
```

Reply:
That's weird, I don't see any other delete of `uncompressed_buffer`. I might have missed something, but from what I see, `uncompressed_buffer` is allocate in `load_blosc_file` (line 36) and returned. Then you passed it to the adaptor and after `load_blosc` exits, only the adaptor has a reference on the underlying buffer.